From ad4ec61969a3dbea5f1371e6d2697759524fb5e1 Mon Sep 17 00:00:00 2001
From: Andi Kleen <andi@firstfloor.org>
Date: Mon, 29 Mar 2021 08:17:46 -0700
Subject: [PATCH 23/39] Compiler attributes: Disable __flatten with LTO

Using __flatten causes a simple gcc 10 LTO build not fit into 16GB
anymore. Disable flatten with LTO.

Signed-off-by: Andi Kleen <andi@firstfloor.org>
---
 include/linux/compiler_attributes.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/linux/compiler_attributes.h b/include/linux/compiler_attributes.h
index 10c603dd21f8..48a004dea9f0 100644
--- a/include/linux/compiler_attributes.h
+++ b/include/linux/compiler_attributes.h
@@ -215,7 +215,12 @@
  * gcc: https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html#Common-Function-Attributes
  * clang: https://clang.llvm.org/docs/AttributeReference.html#flatten
  */
+#ifndef CONFIG_LTO_GCC
 # define __flatten			__attribute__((flatten))
+#else
+/* Causes very large memory use with gcc 10 in LTO mode */
+# define __flatten
+#endif
 
 /*
  * Note the missing underscores.
-- 
2.30.2

